service/src/main/java/org/ohdsi/gisservice/security/sso/SsoProvider.java

```diff
@@ -0,0 +1,76 @@
+package org.ohdsi.gisservice.security.sso;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class SsoProvider {
+
+    @Value("${security.userInfoUri}")
+    private String userInfoUri;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    public Authentication getAuthentication(String token) {
+
+//        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity request = new HttpEntity<>(resolveRequestHeaders(token));
+        ResponseEntity<JsonNode> response;
+
+        try {
+            response = restTemplate.exchange(userInfoUri, HttpMethod.GET, request, JsonNode.class);
```

Comment:
Just to clarify, why `RestTemplate` rather than the Feign?

Reply:
The idea is to later move the Sso related classes into shared component and be able to use them from other projects. Therefore would not like to obligate others to use Feign